sdk/typescript/src/providers/json-rpc-provider.ts

```diff
@@ -523,24 +550,28 @@ export class JsonRpcProvider extends Provider {
       );
     }
   }
-
+ 
   async getTransactionWithEffectsBatch(
     digests: TransactionDigest[]
   ): Promise<SuiTransactionResponse[]> {
-    const requests = digests.map((d) => ({
-      method: 'sui_getTransaction',
-      args: [d],
-    }));
     try {
+      const requests = digests.map((d) => {
+        if (!isValidTransactionDigest(d, "base64") && !isValidTransactionDigest(d, "base58")) {
```

Comment:
```suggestion
        if (!isValidTransactionDigest(d, "base58")) {
```
We have released `0.18.0` on DevNet, which uses `base58`, so we no longer need this check.


@joyqvq do you mind cleaning up the pre-0.18.0 code?